src/lavinmq/amqp/queue/stream_queue_message_store.cr

```diff
@@ -88,6 +92,23 @@ module LavinMQ::AMQP
         {msg_offset, segment, pos}
       end
 
+      private def offset_index_lookup(offset) : UInt32
+        seg = @segments.first_key
+        case offset
+        when Int
+          @offset_index.each do |seg_id, first_seg_offset|
+            break if first_seg_offset >= offset
```

Comment:
if `first_seg_offset` matches the searched for offset we should give that segment, not the full previous segment, right? 
```suggestion
            break if first_seg_offset > offset
```